include/drivers/uart.h

```diff
@@ -157,13 +157,14 @@ enum uart_rx_stop_reason {
 	UART_BREAK = (1 << 3),
 };
 
-/** @brief Backward compatibility defines, deprecated */
-#define UART_ERROR_BREAK UART_BREAK
 #define LINE_CTRL_BAUD_RATE (1 << 0)
-#define LINE_CTRL_RTS UART_LINE_CTRL_RTS
-#define LINE_CTRL_DTR UART_LINE_CTRL_DTR
-#define LINE_CTRL_DCD UART_LINE_CTRL_DCD
-#define LINE_CTRL_DSR UART_LINE_CTRL_DSR
+
+/** @brief Backward compatibility defines, deprecated */
+#define UART_ERROR_BREAK __DEPRECATED_MACRO UART_BREAK
+#define LINE_CTRL_RTS __DEPRECATED_MACRO UART_LINE_CTRL_RTS
+#define LINE_CTRL_DTR __DEPRECATED_MACRO UART_LINE_CTRL_DTR
+#define LINE_CTRL_DCD __DEPRECATED_MACRO UART_LINE_CTRL_DCD
+#define LINE_CTRL_DSR __DEPRECATED_MACRO UART_LINE_CTRL_DSR
```

Comment:
Do we have this definitions used after this patch applied? Probably we can remove them?

Reply:
My understanding is that [deprecated API]( https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#deprecated) needs to remain available for two releases after deprecation.  I don't know when the deprecation was documented, but it wasn't "marked" until this PR.  I also don't know if this was "stable" API; if not it could be removed now.

Reply:
I would just rename them, let's see others opinions here.

Reply:
I think what @pabigot has done here is the proper way to go about it.